Esri/arcgis-maps-sdk-swift-samples
# [Update] Add status messages to samples

## Description

This PR addresses specific feedback from Nick on the initial testing for 200.1.0 sample viewer TestFlight app to add status messages to the `Generate offline map` and `Download vector tiles to local cache` samples:
>- [ ] Generate offline map
>> Would be good to see the status messages here, in the progress view or top status bar.

>- [ ] Download vector tiles to local cache
>> Would be good to see the status messages here.

## Linked Issue(s)

- `swift/issues/3931`

## How To Test
- Generate offline map in `Generate offline map` sample and verify that status text is updated when offline map is generated.
- Download vector tiles in `Download vector tiles to local cache` sample and verify that the status is text is updated when the tiles are downloaded.

## Screenshots
Before/After
<details>
  <summary>Generate offline map</summary>
   <img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/06fe4c0d-8750-4f83-bd39-c30c705efa2d" width="300">    <img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/d5432daf-0e14-41a7-8608-7d1da3a7ec7a" width="300">
  </details>
  
  <details>
  <summary>Download vector tiles to local cache</summary>
 <img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/d9be0102-45a8-4437-8bf7-6096c5d60a2a" width="300">    <img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/bad3d115-a4eb-4140-85c3-cd3098f8ee13" width="300">
  </details>

## To Discuss

- Should the progress views include status text as well?
<details>
  <summary>Progress views</summary>
   <img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/2c139ec0-5a82-450e-9ea1-570e13966a71" width="300">    <img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/ce9d99fc-794e-4d49-a141-9231d4c010be" width="300">
  </details>
